packages/plugins/other/visitor-plugin-common/src/utils.ts

```diff
@@ -495,3 +497,26 @@ function clearOptional(str: string): string {
 function stripTrailingSpaces(str: string): string {
   return str.replace(/ +\n/g, '\n');
 }
+
+const isOneOfTypeCache = new WeakMap<GraphQLNamedType, boolean>();
+export function isOneOfInputObjectType(
+  namedType: GraphQLNamedType | null | undefined
+): namedType is GraphQLInputObjectType {
+  if (!namedType) {
+    return false;
+  }
+  let isOneOfType = isOneOfTypeCache.get(namedType);
+
+  if (isOneOfType !== undefined) {
+    return isOneOfType;
+  }
+
+  isOneOfType =
+    isInputObjectType(namedType) &&
+    ((namedType as unknown as Record<'isOneOf', boolean | undefined>).isOneOf ||
```

Comment:
In the GraphQL version that will ship stable `oneOf` the input object type will actually have a boolean `isOneOf` boolean value.